src/algebra/regular.lean

```diff
@@ -108,6 +109,58 @@ is_regular_mul_and_mul_iff.mp ⟨ab, ba⟩
 
 end semigroup
 
+section monoid
+
+variable [monoid R]
+
+/--  Any power of a left-regular element is left-regular. -/
+lemma is_left_regular.pow (n : ℕ) (rla : is_left_regular a) : is_left_regular (a ^ n) :=
```

Comment:
It would be cool if you could prove this using `function.injective.iterate` - ideally we'd have an `iterate_mul_left` lemma that says left-multiplication by a power is the same as iterated left-multiplication.

Reply:
I've added the lemma I suggested in #6359

Reply:
This would be nice, thanks!  I left a comment on your PR!

Reply:
Eric, I saw that you found the lemmas in a different file and I used them to give a different proof of the lemmas about injectivity of powers: is this what you had in mind?

Thanks!